src/AppInstallerCLICore/Workflows/WorkflowBase.cpp

```diff
@@ -276,8 +313,60 @@ namespace AppInstaller::CLI::Workflow
                 latestVersion->GetProperty(PackageVersionProperty::Id),
                 latestVersion->GetProperty(PackageVersionProperty::Version),
                 GetMatchCriteriaDescriptor(searchResult.Matches[i]),
-                searchResult.Matches[i].SourceName
+                sourceIsComposite ? static_cast<std::string>(latestVersion->GetProperty(PackageVersionProperty::SourceName)) : ""s
+            });
+        }
+
+        table.Complete();
+
+        if (searchResult.Truncated)
+        {
+            context.Reporter.Info() << '<' << Resource::String::SearchTruncated << '>' << std::endl;
+        }
+    }
+
+    void ReportListResult::operator()(Execution::Context& context) const
+    {
+        auto& searchResult = context.Get<Execution::Data::SearchResult>();
+
+        Execution::TableOutput<5> table(context.Reporter,
+            {
+                Resource::String::SearchName,
+                Resource::String::SearchId,
+                Resource::String::SearchVersion,
+                Resource::String::AvailableHeader,
+                Resource::String::SearchSource
             });
+
+        for (const auto& match : searchResult.Matches)
+        {
+            auto installedVersion = match.Package->GetInstalledVersion();
+
+            if (installedVersion)
+            {
+                auto latestVersion = match.Package->GetLatestAvailableVersion();
+                bool updateAvailable = match.Package->IsUpdateAvailable();

```

Comment:
>updateAvailable = match.Package->IsUpdateAvailable(); [](start = 21, length = 53)

This updatability only considers version, are we considering "applicable updates" in the future?

Reply:
I don't think that `list` should.  While upgrade may not automatically "change lanes", I'm sure that we will provide the ability to force it to do so.  If we don't even show that a newer version is available, then how would the user even know that they were falling behind?

---
In reply to: [507683891](https://github.com/microsoft/winget-cli/pull/598#discussion_r507683891) [](ancestors = 507683891)